RoaringBitmap/src/main/java/org/roaringbitmap/BitSetUtil.java

```diff
@@ -1,24 +1,55 @@
 package org.roaringbitmap;
 
 
+import org.roaringbitmap.buffer.ImmutableRoaringBitmap;
+
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.util.BitSet;
 
 
 /***
- *
  * This class provides convenience functions to manipulate BitSet and RoaringBitmap objects.
  *
  */
 public class BitSetUtil {
-  // todo: add a method to convert a RoaringBitmap to a BitSet using BitSet.valueOf
-
   // a block consists has a maximum of 1024 words, each representing 64 bits,
   // thus representing at maximum 65536 bits
-  public static final int BLOCK_LENGTH = BitmapContainer.MAX_CAPACITY / Long.SIZE; //
-  // 64-bit
-  // word
+  public static final int BLOCK_LENGTH = BitmapContainer.MAX_CAPACITY / Long.SIZE;
+
+  /**
+   * Converts an immutable roaring bitmap to JDK bit set. It returns one bit set if bitmap contains
+   * values under Integer.MAX_VALUE only, greater values if present are returned in second bit set
+   * shifted to be non-negative.
+   *
+   * @param bitmap original bitmap
+   * @return bit sets equivalent to roaring bitmap
+   */
+  public static BitSet[] bitSetOf(ImmutableRoaringBitmap bitmap) {
+    // TODO could be optimized
+    // for RunContainer use:
+    //    bitSet.set(from, to);
+    // for BitmapContainer use:
+    // MappeableBitmapContainer bc = (MappeableBitmapContainer)
+    //    bitmap.getContainerPointer().getContainer();
+    //    BitSet bitSet2 = BitSet.valueOf(bc.toLongArray());
+    //    bitSet.and(bitSet2);
+    BitSet bitSetPositive = new BitSet();
+    BitSet bitSetNegative = new BitSet();
+    PeekableIntIterator it = bitmap.getIntIterator();
+    while (it.hasNext()) {
+      int value = it.next();
+      if (value >= 0) {
+        bitSetPositive.set(value);
+      } else {
+        bitSetNegative.set(value - Integer.MIN_VALUE);
+      }
+    }
+    if (bitSetNegative.isEmpty()) {
+      return new BitSet[]{bitSetPositive};
+    }
```

Comment:
What if bitSetPositive is empty?

It seems simpler to just return both bitsets...

Reply:
This would be much better than throw an exception when if someone wants to convert "negative" (as RoaringBitmap treats them as unsigned int), then he must do some expensive work before - split bitmap and reset high bit for negative values.